private_gpt/components/vector_store/vector_store_component.py

```diff
@@ -121,6 +121,45 @@ def __init__(self, settings: Settings) -> None:
                         collection_name="make_this_parameterizable_per_api_call",
                     ),  # TODO
                 )
+
+            case "milvus":
+                try:
+                    from llama_index.vector_stores.milvus import (  # type: ignore
+                        MilvusVectorStore,
+                    )
+                except ImportError as e:
+                    raise ImportError(
+                        "Milvus dependencies not found, install with `poetry install --extras vector-stores-milvus`"
+                    ) from e
+
+                if settings.milvus is None:
+                    logger.info(
+                        "Milvus config not found. Using default settings.\n"
+                        "Trying to connect to Milvus at ./milvus_llamaindex.db "
+                        "with collection 'make_this_parameterizable_per_api_call'."
+                    )
+
+                    self.vector_store = typing.cast(
+                        BasePydanticVectorStore,
+                        MilvusVectorStore(
+                            dim=settings.embedding.embed_dim,
+                            collection_name="make_this_parameterizable_per_api_call",
+                            overwrite=True,
+                        ),
```

Comment:
I'm not sure if this is a good idea. I know that we have similar flow with Qdrant.
Does it make sense to initialize the vector store if there is no settings?

Reply:
No problem, thanks for the comments.

Reply:
I think having a default case is necessary, for it at least support all users to able to begin the project.

Reply:
Since the configuration are in `settings.yaml`, and will therefore always be loaded, this flow would only occur if a user deletes `MilvusDB` section from `settings.yaml`. That is why I asked about it.

Reply:
You have the point, maybe we should ignore all the "settings is None" case, since users deliberately delete the configs might indicate that they don't wish to use that particular vector db.

Reply:
Should I change the flow in this pr?

Reply:
LGTM
Thanks!